ambry-account/src/test/java/com/github/ambry/account/HelixAccountServiceTest.java

```diff
@@ -606,6 +602,48 @@ public void testAccountUpdateConsumer() throws Exception {
   }
 
   /**
+   * Tests the background updater for updating accounts from remote. Because of using the {@link MockHelixPropertyStore},
+   * everything is single-threaded except for the background updater, so after the {@link HelixAccountService} starts,
+   * it should have made at least one get call to the {@link MockHelixPropertyStore}, and then all the get calls should
+   * be made by the updater, not from the listener. This test sets the updating interval to 1ms, and ensures that after
+   * 2ms the get calls to the {@link MockHelixPropertyStore} is greater than one.
+   * @throws Exception
+   */
+  @Test
+  public void testBackgroundUpdater() throws Exception {
+    helixConfigProps.setProperty(
+        HelixPropertyStoreConfig.HELIX_PROPERTY_STORE_PREFIX + "account.service.polling.interval.ms", "1");
+    vHelixConfigProps = new VerifiableProperties(helixConfigProps);
+    storeConfig = new HelixPropertyStoreConfig(vHelixConfigProps);
+    MockHelixAccountServiceFactory mockHelixAccountServiceFactory =
+        new MockHelixAccountServiceFactory(vHelixConfigProps, new MetricRegistry(), notifier, shouldUseMockHelixStore);
+    accountService = mockHelixAccountServiceFactory.getAccountService();
+    Thread.sleep(2);
```

Comment:
let's make these tests deterministic and use latches.

We can also use the `numThreadsByName` method in `TestUtils` to check for existence of the thread.

Reply:
Fixed.